### PR TITLE
ASoC: SOF/Intel: clarify SPDX license with GPL-2.0-only

### PR DESCRIPTION
**Pull request by plbossart**
Remove the ambiguity with GPL-2.0 and use an explicit GPL-2.0-only
tag.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

**Comment by plbossart** [comment]
@dbaluta please double-check that you are ok with the imx/ change, thanks!

File: include/sound/sof/control.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
```
Comment (lyakh) [inline include/sound/sof/control.h:1]: @plbossart what does GPL-v2.0-only or BSD-3 mean?

Comment (RanderWang) [reply include/sound/sof/control.h:1]: I have a same question

Comment (kv2019i) [reply include/sound/sof/control.h:1]: @lyakh @RanderWang The canonical/original GPL license always had the "or later" in license text. Linux kernel specifically omitted this text. So I believe this change is to make it clear the Linux kernel version of the license is referred to here. E.g. https://spdx.org/licenses/GPL-2.0-only.html
I don't believe it has any impact dual-licensing of code, just to specify which version of GPL more exactly.

Comment (plbossart) [reply include/sound/sof/control.h:1]: yes, this to clarify that there is no permission to use this code with a GPL-2.0-or-later.
Don't ask me why GPL2.0 can be interpreted in two ways, I am just doing the correction suggested by smarter people than me.

Comment (lyakh) [reply include/sound/sof/control.h:1]: @kv2019i @plbossart thanks for the clarification!

**Comment by plbossart** [comment]
@dbaluta ping for imx changes, thanks!

**Comment by dbaluta** [comment]
all good for imx. Again, missed the notification. Pierre just send me an email if I dont answer in a reasonable time.

**Pull request merged**